### DISCUSSION TITLE
Feature request: support attaching source code files

### DISCUSSION THREAD

**Discussion by guyharris** [community/community#65941] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Product Feedback

### Body

Please allow attaching source code files to issues.

Or, rather, allow *directly* attaching them, rather than, for example, renaming file.c to file.txt and then attaching with a note that somebody who downloads it will need to save it as file.c or rename it to file.c.

People who file issues might want to attach a test program to reproduce the issue (see, for example, https://github.com/the-tcpdump-group/libpcap/issues/1219), and people responding to issues might want to attach a test program to request that the user who submitted the issue compile the test program, run it, and put the results in a reply.

**Comment by xexyl** [comment]
Indeed. It is kind of hard to believe that one cannot directly upload a C file or a header file. Or a java file. Or a flex or bison file. Or many others. There are countless reasons to allow this and absolutely no reasons to not allow it.

**Comment by lcn2** [comment]
**PLEASE** GitHub ‼️

This is a place where people want / need to discuss source code.  We need to be able to drop in source code files directly.

Consider what had to happen with this comment: [GH-issuecomment-2571699488](https://github.com/xexyl/jparse/issues/33#issuecomment-2571699488)

The user has to **trick** the interface into attaching some source code by compressing it, and force others to have to download and then decompress the attached source code.

**PLEASE** allow us to attach source code files directly to comments!

**Comment by SirWumpus** [comment]
I've had to do similar of late https://github.com/ioccc-src/mkiocccentry/issues/1056#issuecomment-2569635835, take a test C source file and rename it just so I can upload it.  Not being able to upload source files into comments seems counter to the purpose of a website aimed at aiding developers.  We have enough to do, we should not have to hoop jump to get work done.

`...only now we do it under the noses of twenty trained Alliance Feds. And that makes 'em look all manner of stupid.` - Mal, Firefly

**Reply by xexyl** [reply]
> I've had to do similar of late https://github.com/ioccc-src/mkiocccentry/issues/1056#issuecomment-2569635835, take a test C source file and rename it just so I can upload it.  Not being able to upload source files into comments seems counter to the purpose of a website aimed at aiding developers.  We have enough to do, we should not have to hoop jump to get work done.
> 
> 
> 
> `...only now we do it under the noses of twenty trained Alliance Feds. And that makes 'em look all manner of stupid.` - Mal, Firefly 
> 
> 

Oddly yesterday I had to gzip a C file because adding .txt didn't work. 

Today I had to include a header file too so I made it a tarball but of course some kinds of tarballs are not allowed either. 

It does seem very counterintuitive to what the website is supposed to be doing. 

It's pretty boggling.

**Reply by guyharris** [reply]
And it's not just source code; at least for [tcpdump](https://github.com/the-tcpdump-group/tcpdump), it's useful to be able to attach network packet captures. See https://github.com/the-tcpdump-group/tcpdump/issues/1331#issuecomment-2995629461. That might also be useful for bug reports on software that uses network traffic.

**Reply by SirWumpus** [reply]
Its probably worth having the repo owner specify which file extensions they are willing to directly work with.  We're a `C project` so `.c` `.h` `.mk` might be acceptable.  We're a `tcpdump` project so `.pkt` or `.dump` or `.xyzzy` are our preferred extensions.  Everyone is happy to make a fuss about personal pronouns, but when it comes to useful file extensions for a project's bug reports, we have to fart about with workarounds.

`Oh drat these computers, they're so naughty and so complex.` - Marvin The Martian, Looney Tunes
